azurelinuxagent/common/protocol/wire.py

```diff
@@ -661,6 +694,23 @@ def stream(self, uri, destination, headers=None, use_proxy=None):
 
         return success
 
+    def fetch_content_and_etag(self, uri, headers=None, use_proxy=None, decode=True):
```

Comment:
NIT: Can you rename this function to be more clear as to what it does - `fetch_content_and_etag_if_modified`.
Also can you make it private?

Reply:
Also in this scenario (as per the code), `etag` will be not None only if `not_modified` is True right? Can't we just use this knowledge to deduce if it was modified or not instead of passing the `not_modified` variable?

Reply:
Also, shouldn't the name be `modified` instead of `not_modified`? It just seems counter intuitive to see `not_modified=False` when it actually means `modified=False`